### ISSUE TITLE
Reloading column filters

### ISSUE THREAD

**Issue by jimmccarthy** [free-jqgrid/jqGrid#103]
I have been struggling with trying to save and reload column filters for the report toolbar.  I am doing the following:

I pull out and save the current filters using: jqGrid('getGridParam','postData').filters
To set the filters when rebuilding a report I set a postData param when the grid is instantiated.  I also set search to 'true'.  That applies the filter to the report when it is created, the problem is that it does not fill in the search values in the actual input controls on the toolbar.

I have tried to work around this by setting searchoptions.DefaultValue on colModel attributes by parsing the JSON string returned from getGridParam on postData, but it seems like there should be a better way to do this.  Am I missing something?

I looked through the code and could not see where calling setGridParam would actually update the toolbar input values.  Seems like if postData is set when the grid is instantiated it should set the toolbar filter input values as well.

Any help is appreciated,

thanks,

Jim

**Comment by OlegKi** [comment]
Hi Jim,

the current implementation of `filterToolbar` don't fill the filter. Only advanced searching dialog do this. The main problem: the filter could be more complex as the `fillterToolbar` can display.

I would recommend you to try to use `refreshSerchingToolbar` function which I described originally for [the answer](http://stackoverflow.com/a/6884755/315935) and used later for [the demo](http://www.ok-soft-gmbh.com/jqGrid/ColumnChooserAndLocalStorage1.htm) created for [the answer](http://stackoverflow.com/a/8436273/315935). It fills filter toolbar in case of usage *simple filters*. What I mean: I wrote the function `refreshSerchingToolbar` 4 years ago. At the time the filter toolbar had no option `searchOperators: true`. So the function `refreshSerchingToolbar` uses only the first element of `searchoptions.sopt` or `defaultSearch` option of `filterToolbar` to fill the filter. The function `refreshSerchingToolbar` can be improved for the more common case, but the old version still good work in the most situations. Probably it's enough for your purpose.

By the way, I hope you know that free jqGrid 4.9 is published and you use it.

Best wishes
Oleg
